src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides should be greater than 0.');
+    }
+
+    const sides = [a, b, c].sort((x: number, y: number) => {
+      return y - x;
+    });
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides.',
+      );
+    }
+  }
+
+  getArea(): number {
```

Comment:
[CODE STYLE] - The `getArea` function should not accept any parameters. All necessary values are already available as object fields.